library/ssl_tls12_client.c

```diff
@@ -2010,65 +2010,6 @@ static int ssl_write_encrypted_pms( mbedtls_ssl_context *ssl,
 #endif /* MBEDTLS_KEY_EXCHANGE_RSA_ENABLED ||
           MBEDTLS_KEY_EXCHANGE_RSA_PSK_ENABLED */
 
-#if defined(MBEDTLS_KEY_EXCHANGE_DHE_RSA_ENABLED) ||                       \
```

Comment:
I do not see any imperative reason why we should have to change some TLS 1.2 library code in this PR. Thus please reverse this change of the TLS 1.2 library code and if some tests in some configuration fail let's try to find another way to make them pass again.

Reply:
Got it. will move it to another PR.